Feature/adf 1424/acl section on destination selector

rgomeztao opened the pull request:
coming from [https://oat-sa.atlassian.net/browse/ADF-1424](https://oat-sa.atlassian.net/browse/ADF-1424)

changelog:
* adds acl radio buttons to destination selector

jsconan commented on src/destination/selector.js:
```diff
@@ -108,6 +112,10 @@ export default function destinationSelectorFactory($container, config) {
                 }
             };
 
+            var getSelectedACLTransferMode = function getSelectedACLTransferMode() {
+                return $('input[name="acl-mode"]:checked').val();
+            };
```
**thought:** New code can be written in ES6 in this repo. Actually, when possible, we can also convert existing code to ES6. Anyway not a blocker.

**suggestion:**
```suggestion
            const getSelectedACLTransferMode = () => $('input[name="acl-mode"]:checked').val();
```

rgomeztao replied:
updated, thanks!

jsconan commented on src/destination/tpl/selector.tpl:
```diff
@@ -4,9 +4,18 @@
         {{/if}}
         <div class="selector-container"></div>
+        {{#if showACL}}
+        <div class="permissions-settings-container">
+            <p class="section-title">{{__ 'Permission settings'}}</p>
+            <input type="radio" id="acl-keep-original" {{#equal aclTransferMode 'acl.keep.original' }}checked {{/equal}}name="acl-mode" value="acl.keep.original" />
+            <label for="acl-keep-original">{{__ 'Keep the original permissions'}}</label><br>
+            <input type="radio" id="acl-use-destination" {{#equal aclTransferMode 'acl.use.destination' }}checked {{/equal}}name="acl-mode" value="acl.use.destination" />
+            <label for="acl-use-destination">{{__ 'Inherit the permission of the parent folder'}}</label>
+        </div>
```
**issue:** There are severals issues in this code:
- The component does not uses a form despite it manages input fields. While this is ok since the component manage the inputs by itself, it may impact the container if it is also a form as the name fields will be part of it silently. Suggestion: don't use `name`, prefer `data-name` or similar.
- The labels are identified, but the ID is not guaranteed to be unique within the page, leading to a possible collision. Suggestion: don't use ID, prefer nested label instead `<label><input /></label>`, or use dynamic unique ID if the nested label cannot apply

rgomeztao replied:
updated. Unfortunately, removing name from the inputs will remove the relation between the button set. however,  thats something we may fix with some javascript :)

jsconan replied:
Correct, the name allows grouping the buttons. Please ignore my previous comment, the name should be present.

jsconan reviewed (commented):
It looks nice. However, I have a few remarks. Please see below.

As a side note, could I suggest fixing the visual unit test too?
- you can add this to the file `test.html`, in the stylesheet:
    ```css
      div#qunit {
          position: relative;
      }
    ```
- you could also add the config for the visual test:
    ```javascript
            ...,
            showACL: true,
            aclTransferMode: 'acl.keep.original'
    ```

jsconan commented on src/destination/tpl/selector.tpl:
```diff
@@ -4,9 +4,18 @@
+        {{#if showACL}}
+        <div class="permissions-settings-container">
+            <p class="section-title">{{__ 'Permission settings'}}</p>
+            <input type="radio" class="acl-mode-radio acl-keep-original" {{#equal aclTransferMode 'acl.keep.original' }}checked {{/equal}} value="acl.keep.original" />
+            <label for="acl-keep-original">{{__ 'Keep the original permissions'}}</label><br>
+            <input type="radio" class="acl-mode-radio acl-use-destination" {{#equal aclTransferMode 'acl.use.destination' }}checked {{/equal}} value="acl.use.destination" />
+            <label for="acl-use-destination">{{__ 'Inherit the permission of the parent folder'}}</label>
```
**issue:** The label is not linked with the input. Actually, the `for` attribute needs an ID.
See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#attributes

**suggestion:** Use dynamic ID generated from the component
```suggestion
            <input id="acl-keep-original{{uniqId}}" type="radio" class="acl-mode-radio acl-keep-original" {{#equal aclTransferMode 'acl.keep.original' }}checked {{/equal}} value="acl.keep.original" />
            <label for="acl-keep-original{{uniqId}}">{{__ 'Keep the original permissions'}}</label><br>
            <input id="acl-use-destination{{uniqId}}" type="radio" class="acl-mode-radio acl-use-destination" {{#equal aclTransferMode 'acl.use.destination' }}checked {{/equal}} value="acl.use.destination" />
            <label for="acl-use-destination{{uniqId}}">{{__ 'Inherit the permission of the parent folder'}}</label>
```

jsconan reviewed (changes requested):
Almost there, still a few things to address, unfortunately.

rgomeztao commented on src/destination/selector.js:
```diff
@@ -85,13 +89,17 @@ export default function destinationSelectorFactory($container, config) {
             var $component = this.getElement();
 
+            $('.acl-mode-radio').change(function () {
+                $('.acl-mode-radio').not(this).prop('checked', false);
```
fixed, ill keep in mind for the future :)

rgomeztao commented on src/destination/selector.js:
```diff
@@ -108,6 +116,8 @@ export default function destinationSelectorFactory($container, config) {
                 }
             };
 
+            const getSelectedACLTransferMode = () => $('input.acl-mode-radio:checked').val();
```
fixed, thanks!

rgomeztao commented on test/destination/selector/test.css:
```diff
@@ -0,0 +1,3 @@
+div#qunit {
+    position: relative;
+}
```
fixed, well i took inspiration from here https://github.com/oat-sa/tao-core-ui-fe/blob/e3bff77a61f9cac5345acf49c269434c42ec0b30/test/contextualPopup/test.css

jsconan replied:
Ok got it. But it was incomplete because the CSS also need to be loaded, like here: https://github.com/oat-sa/tao-core-ui-fe/blob/e3bff77a61f9cac5345acf49c269434c42ec0b30/test/contextualPopup/test.js#L1

jsconan commented on src/destination/selector.js:
```diff
@@ -85,13 +89,17 @@ export default function destinationSelectorFactory($container, config) {
         .setTemplate(selectorTpl)
-        .on('init', function() {
+        .on('init', function () {
             this.render($container);
```
**suggestion:** the `uniqId` needs to be set in the config
```suggestion
            this.config.uniqId = uuid(); // need to add this too: `import uuid from 'lib/uuid';`
            this.render($container);
```

rgomeztao replied:
> import uuid from 'lib/uuid';

fixed and added the import line to imports area :) thanks!

jsconan commented:
Please also fix the lint issue at this line: https://github.com/oat-sa/tao-core-ui-fe/pull/555/files#diff-5f3fae7476c943fe8d4adcb6ee604ea84f8bb79fb5750d087ba85bdbf7ed0172R107

You can fix it easily by removing the useless var declaration (the function is self declarative)

rgomeztao commented:
> 

fixed :thinking:

i have chnaged the var for a const, just to be consistent with the next line, and changed the function by an arrow function
 
and apparently yes, it seems that neither lintering or prettier have not been working as expected on this project, usually i run them from command line, but on these project both were giving false negatives report :thinking:

jsconan reviewed (approved):
It works nicely. Good job.

Regarding the names of the radio buttons, you are right, they should be present. Please ignore my previous comment.

**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code respects code style rules
- [x] New code respects best practices
- [ ] ~~New code is not subject to concurrency issues (if applicable)~~
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
